server/irmaserver/sessions.go

```diff
@@ -138,6 +139,10 @@ func (s *memorySessionStore) deleteExpired() {
 				s.conf.Logger.WithFields(logrus.Fields{"session": session.requestorToken}).Infof("Session expired")
 				session.markAlive()
 				session.setStatus(irma.ServerStatusTimeout)
+				if handler := session.handler; handler != nil {
+					session.handler = nil
+					go handler(session.result)
+				}
```

Comment:
I saw that `session.setStatus` is also called to process cancellation. Cancellation can also be done using the `CancelSession` API function. This function is for example used by the `requestorserver` to cancel a session. When using the API function, the `sessionMiddleware` of `irmaserver` is also bypassed, so this issue will occur in that case too right?

I think that variant of the same bug should be resolved too then.